elasticsearch/src/http/transport.rs

```diff
@@ -422,6 +461,70 @@ impl Transport {
         if let Some(q) = query_string {
             request_builder = request_builder.query(q);
         }
+        Ok(request_builder)
+    }
+
+    /// Creates an asynchronous request that can be awaited
+    pub async fn send<B, Q>(
+        &self,
+        method: Method,
+        path: &str,
+        headers: HeaderMap,
+        query_string: Option<&Q>,
+        body: Option<B>,
+        timeout: Option<Duration>,
+    ) -> Result<Response, Error>
+    where
+        B: Body,
+        Q: Serialize + ?Sized,
+    {
+        // Threads will execute against old connection pool during reseed
+        if self.conn_pool.reseedable() {
+            // Set as reseeding prevents another thread from attempting
+            // to reseed during es request and reseed
+            self.conn_pool.reseeding();
```

Comment:
My understanding of the current implementation is because the `reseedable` is a `read` lock and the `reseeding` is a `write` lock that multiple threads are unlikely to enter the reseeding subroutine at the same time (though its possible because the `reseedable` and `reseeding` flags are not atomic). The consequence is two reseeds (that are set behind a write lock) setting the connections Vector and then setting it again. I think this trade off is ok vs the added complexity of creating a seperate thread on `MultiConnectionPool` to monitor reseeding. Though I am open to suggestions if one is more favorable than the other.